File: py/solns/twoSum/twoSum.py
class Solution:
    @staticmethod
    def naive(nums,target):
        length = len(nums)
        for i in range(length):
            for j in range(i+1,length):
                if nums[i]+nums[j] == target:
                    return [i,j]
    
    @staticmethod
    def hash(nums,target):
        mapNum2Idx = {}
        for i in range(len(nums)):
            dif = target - nums[i]
            if dif in mapNum2Idx:
                return [i,mapNum2Idx[dif]]
            mapNum2Idx[nums[i]] = i

File: cpp/example/src/TwoSum/twoSum.cpp
#include "twoSum.h"
#include <vector>
#include <iostream>

using namespace std;

vector<int> twoSum::naive(vector<int>& nums,int target){
    vector<int> v;
    int length = nums.size();
    for (int i=0;i<length;i++){
        for (int j=i+1;j<length;j++){
            if (nums[i]+nums[j] == target){
                v.push_back(i);
                v.push_back(j);
                return v;
            }
        }
    }
}

vector<int> twoSum::hash(vector<int>& nums,int target){
    vector<int> v;
    return v;
}

